### PR TITLE
Lnurl pay

### PR DESCRIPTION
**Pull request by roeierez**

### CODE REVIEW

File: lib/bloc/webhooks/webhooks_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        lnurlPayError: lnurlPayError ?? this.lnurlPayError,
+        loading: loading ?? this.loading);
+  }
+}
+
+class WebhooksBloc extends Cubit<WebhooksState> {
+  static const notifierServiceURL = "https://notifier.breez.technology";
+  static const lnurlServiceURL = "https://lnurl.breez.technology";
+
+  final _log = Logger("LnurlPayBloc");
```
Comment (erdemyerebasmaz) [inline lib/bloc/webhooks/webhooks_bloc.dart:59]:
```suggestion
  final _log = Logger("WebhooksBloc");
```

File: lib/bloc/webhooks/webhooks_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        : defaultTargetPlatform == TargetPlatform.android
+            ? "android"
+            : "";
+    if (platform.isEmpty) {
+      throw Exception("Notifications for platform is not supported");
+    }
+    return "$notifierServiceURL/api/v1/notify?platform=$platform&token=$token";
+  }
+
+  String generateUser() {
```
Comment (erdemyerebasmaz) [inline lib/bloc/webhooks/webhooks_bloc.dart:159]: This function is never used and it's better off being an utility function under utils folder.

File: lib/routes/lightning_address/lightning_address_page.dart [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+}
+
+class LightningAddressPageState extends State<LightningAddressPage> {
+  @override
+  void initState() {
+    super.initState();
+    final webhookBloc = context.read<WebhooksBloc>();
+    webhookBloc.refreshLnurlPay();
+    context.read<InputBloc>().trackPayment(null).then((value) {
```
Comment (erdemyerebasmaz) [inline lib/routes/lightning_address/lightning_address_page.dart:31]: What if we receive another payment? How does this work in terms of reliability?

**Review by erdemyerebasmaz — CHANGES_REQUESTED** [review]
Looks good overall, have some nit comments

File: lib/bloc/webhooks/webhooks_bloc.dart (hunk 1 above)
Comment (roeierez) [reply lib/bloc/webhooks/webhooks_bloc.dart]: fixed

File: lib/bloc/webhooks/webhooks_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  }
+}
+
+class WebhooksBloc extends Cubit<WebhooksState> {
+  static const notifierServiceURL = "https://notifier.breez.technology";
+  static const lnurlServiceURL = "https://lnurl.breez.technology";
+
+  final _log = Logger("LnurlPayBloc");
+  final BreezSDK _breezSDK;
+  final ServiceInjector injector;
```
[the remark this answers is not in the document]
Comment (roeierez) [reply lib/bloc/webhooks/webhooks_bloc.dart]: fixed

File: lib/bloc/webhooks/webhooks_bloc.dart (hunk 2 above)
Comment (roeierez) [reply lib/bloc/webhooks/webhooks_bloc.dart]: Right, sneaked by mistake.

File: lib/routes/lightning_address/lightning_address_page.dart (hunk 3 above)
Comment (roeierez) [reply lib/routes/lightning_address/lightning_address_page.dart:31]: We don't really have a way to associate a specific payment. Worst case (which is very rare) we will exit this screen on payment and show the payment receive screen (which sounds OK to me).

Comment (erdemyerebasmaz) [reply lib/routes/lightning_address/lightning_address_page.dart:31]: Perhaps we can pass a flag here, `isLnurlPay` and check if the next received payment is a ln payment on trackPayment as we have payment details there. Either way, it's OK for me.

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM

Have one optional request: https://github.com/breez/c-breez/pull/759#discussion_r1463356717

File: lib/routes/lightning_address/lightning_address_page.dart (hunk 3 above)
Comment (roeierez) [reply lib/routes/lightning_address/lightning_address_page.dart:31]: The trackPayment always notifies only for Lnpayment so we don't need the flag.

**Pull request merged**